Dockerfile

```diff
@@ -0,0 +1,16 @@
+# Perform common operations, dependency installation etc...
+FROM python:3.9.0-buster as base
+RUN pip install poetry
+WORKDIR /DevOps-Course-Starter
+COPY pyproject.toml /DevOps-Course-Starter/
+COPY . /DevOps-Course-Starter/
+
+# Configure for production
+FROM base as production
+RUN poetry install  --no-dev
```

Comment:
This is a nice way to separate out the install commands to take advantage of `--no-dev `, but this ordering means that each docker build will need to go through the `poetry install` step if any python file has been updated, which can be one of the slowest steps. It would be worth thinking about how we could avoid that